File: exercism/go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert return Raindrops sound string if number matches
func Convert(input int) string {

	// Cannot use map here as it is unordered
	sounds := []struct {
		number int
		sound  string
	}{
		{3, "Pling"},
		{5, "Plang"},
		{7, "Plong"},
	}

	var ret string
	for _, v := range sounds {
		if input%v.number == 0 {
			ret += v.sound
		}
	}

	if ret == "" {
		return strconv.Itoa(input)
	}
	return ret

}

File: exercism/r/raindrops/raindrops.R
raindrops <- function(number) {
    # num <- c(3,5,7)
    # sound <- c('Pling', 'Plang', 'Plong')
    # bools <- sapply(num, function(it){number %% it == 0})
    # val <- paste(sound[bools], collapse='')
    # if(val == ''){
    #     return (toString(number))
    # }
    # val    

    sounds <- list("Pling", "Plang", "Plong")
    mask <- c(number %% 3 == 0, number %% 5 == 0, number %% 7 == 0)
    ifelse(any(mask), do.call(paste0, sounds[mask]), as.character(number))
}